### ISSUE TITLE
Reject malformed constraint with unbound KVar parameter

### ISSUE THREAD

**Issue by ranjitjhala** [ucsd-progsys/liquid-fixpoint#274]
see `tests/pos/bad-subst01.fq` (which passes with `--eliminate=some`).

The binder `zogbert` is junk and should not be in the WF of the KVAR as 
it is NOT substituted away at the use sites.

See also LH `tests/pos/zipper.hs` which now (in `no-prop`) fails due to:

```
  crash: SMTLIB2 respSat = Error "line 597 column 3933: unknown constant zcmp"
```

The binder `zcmp` is not in any environment, but appears in 
substitutions `K[foo := zcmp]`. Such a substitution is bogus,
we should remove "foo" from the parameters of `K` and then
remove the substitution too.

(Ideally, such constraints should be rejected...)

**Comment by ranjitjhala** [comment]
Hmm pretty sure I fixed these...

**Comment by ranjitjhala** [comment]
Should be fixed by PR #275
